linkedList.js

```diff
@@ -4,5 +4,79 @@
  * @return {Object} an object exposing methods to be used to manipulate a linked list
  */
 function linkedListGenerator(){
+    let head = null;
+    let tail = null;
+    return {
+        getHead: function(){
+            return head;
+        },
+        getTail: function(){
+            return tail;
+        },
+        add: function(value){
+            let newNode = new Object;
+            newNode.value = value;
+            newNode.next = null;
+            if (!head){
+                head = newNode;
+                tail = newNode;
+            } else {
+                tail.next = newNode;
+                tail = newNode;
+            }
+            return newNode;
+        },
+        get: function(index){
+            let current = head;
 
+            let i = 0;
+            while (current !== null) {
+                if (i === index) {
+                    return current;
+                } else { 
+                    current = current.next;
+                    i++;
+                }
+            }
+            return false;
+         
+        },
+        remove: function(index){
+            let current = this.get(index);
+            let before;
+            if (current === false){
+                return false;
+            }
+            if (index <= 0){
+                head = current.next;
+                before = false;
+            } else if (!this.get(index + 1)) {
+                before = this.get(index - 1);
+                tail = before;
+            } else {
+                before = this.get(index - 1);
+            }
+            before.next = current.next;
```

Comment:
just move this line under line 57. The way that it stand, trying to remove from head will work, but set `false.next` to `current.next` polluting the boolean `false`

Reply:
On second thought, you might need to copy this line in the else if as well. Also, you might as well set the before on line 46 so you don't have to do it in multiple places.